lib/spack/spack/environment/environment.py

```diff
@@ -2165,14 +2050,15 @@ def is_latest_format(manifest):
 
 
 @contextlib.contextmanager
-def deactivate_environment():
+def environment_deactivated():
     """Deactivate an active environment for the duration of the context."""
-    global _active_environment
-    current, _active_environment = _active_environment, None
+    env = active_environment()
     try:
+        deactivate()
         yield
     finally:
-        _active_environment = current
+        # TODO: we don't handle `use_env_repo` here.
```

Comment:
Hm, it can be stored in the env itself. I was also thinking that it wouldn't hurt ot get rid of the flag, and instead put this in the spack.yaml, which is more transparent than remembering whether or not you had this --use-env-repo set enabled or not.

Reply:
but that's for another pr...

Reply:
Fair enough.